src/js/DataTables/EditDialogColumn.js

```diff
@@ -70,597 +119,547 @@ export default class EditDialogColumn extends PureComponent {
+     * @see {@link #placeholder}
+     * @see {@link TextFields/TextField#block}
      */
-    onKeyUp: PropTypes.func,
+    label: PropTypes.string,
 
     /**
-     * An optional function to call when the keydown event is triggered.
+     * An optional placeholder for the text field.
      */
-    onKeyDown: PropTypes.func,
+    placeholder: PropTypes.string,
 
     /**
-     * An optional function to call when the mouseover event is triggered.
+     * Boolean if the edit dialog should become a large dialog. When the dialog is large,
+     * the `title` prop is required.
+     *
+     * A large dialog has a Title followed by the text field, and then a cancel and ok action
+     * buttons below.
      */
-    onMouseOver: PropTypes.func,
+    large: PropTypes.bool,
 
     /**
-     * An optional function to call when the mouseleave event is triggered.
+     * The title to use for the large edit dialog. This prop is required if the `large` prop
+     * is enabled.
      */
-    onMouseLeave: PropTypes.func,
+    title: PropTypes.node,
 
     /**
-     * An optional function to call when the touchstart event is triggered.
+     * An optional `maxLength` to apply to the text field.
+     *
+     * @see {@link TextFields/TextField#maxLength}
      */
-    onTouchStart: PropTypes.func,
+    maxLength: PropTypes.number,
 
     /**
-     * An optional function to call when the touchend event is triggered.
+     * Any children required to render the inline edit icon.
      */
-    onTouchEnd: PropTypes.func,
+    inlineIconChildren: PropTypes.node,
 
     /**
-     * Boolean if the edit dialog should be large.
+     * The icon class name used to render the inline edit icon.
      */
-    large: PropTypes.bool,
+    inlineIconClassName: PropTypes.string,
 
     /**
-     * The title for the large edit dialog. The custom validation changes to required
-     * when the `large` prop is set to true.
+     * Boolean if no inline edit icon should be used.
      */
-    title: (props, propName, component, ...others) => {
-      if (props.large) {
-        return PropTypes.string.isRequired(props, propName, component, ...others);
-      } else {
-        return PropTypes.string(props, propName, component, ...others);
-      }
-    },
+    noIcon: PropTypes.bool,
 
     /**
-     * An optional function to call when the OK button is clicked.
-     * It is called with `(textFieldValue, clickEvent)`. This function
-     * will also be called when a user pressed the enter key.
+     * An optional function to call when the "Ok" button has been clicked. This
+     * is only valid if the edit dialog is `large`.
+     *
+     * The callback will include the current value and the click or keypress event.
+     * ```js
+     * onOkClick(value, event)
+     * ```
+     *
+     * @see {@link #large}
      */
     onOkClick: PropTypes.func,
 
     /**
-     * The label to use for the OK button.
+     * The label to use for the "Ok" button in large dialogs.
```

Comment:
Add documentation notes about the `primary` / `secondary` ok/cancel additions in release notes